### PR TITLE
feat: unit tests for validator_state.py

### PR DESCRIPTION
**Pull request by sergeyWh1te**

### CODE REVIEW

File: tests/utils/test_validator_state.py [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+    assert actual == expected
+
+
+@pytest.mark.unit
+@pytest.mark.parametrize("exit_epoch, expected", [
+    (176720, True),
+    (2 ** 64 - 1, False),
+])
+def test_is_on_exit(exit_epoch, expected):
+    validator_state = ValidatorState(
```
Comment (skhomuti) [inline tests/utils/test_validator_state.py:79]: please create factory for `ValidatorState`, see `tests/factory` files

Comment (skhomuti) [reply tests/utils/test_validator_state.py:79]: and reuse some of them

Comment (sergeyWh1te) [reply tests/utils/test_validator_state.py:79]: Sure!

Comment (sergeyWh1te) [reply tests/utils/test_validator_state.py]: Resolved

File: src/utils/validator_state.py [hunk 2]
```diff
@@ -80,4 +80,7 @@ def calculate_total_active_effective_balance(validators: list[Validator], ref_ep
         if is_active_validator(v, ref_epoch):
             total_effective_balance += int(v.validator.effective_balance)
 
+    if total_effective_balance == 0:
```
Comment (F4ever) [inline src/utils/validator_state.py:83]: This is wrong

File: src/utils/validator_state.py [hunk 3]
```diff
@@ -53,9 +53,9 @@ def is_fully_withdrawable_validator(validator: Validator, epoch: EpochNumber) ->
     https://github.com/ethereum/consensus-specs/blob/dev/specs/capella/beacon-chain.md#is_fully_withdrawable_validator
     """
     return (
-        has_eth1_withdrawal_credential(validator)
-        and EpochNumber(int(validator.validator.withdrawable_epoch)) <= epoch
-        and Gwei(int(validator.balance)) > Gwei(0)
+            has_eth1_withdrawal_credential(validator)
```
Comment (F4ever) [inline src/utils/validator_state.py:56]: remove tabs

File: src/utils/validator_state.py [hunk 4]
```diff
@@ -33,9 +33,9 @@ def is_partially_withdrawable_validator(validator: Validator) -> bool:
     has_max_effective_balance = int(validator.validator.effective_balance) == MAX_EFFECTIVE_BALANCE
     has_excess_balance = int(validator.balance) > MAX_EFFECTIVE_BALANCE
     return (
-        has_eth1_withdrawal_credential(validator)
-        and has_max_effective_balance
-        and has_excess_balance
+            has_eth1_withdrawal_credential(validator)
```
Comment (F4ever) [inline src/utils/validator_state.py:36]: remove tabs

File: src/utils/validator_state.py [hunk 5]
```diff
@@ -2,7 +2,7 @@
     MAX_EFFECTIVE_BALANCE,
     ETH1_ADDRESS_WITHDRAWAL_PREFIX,
     SHARD_COMMITTEE_PERIOD,
-    FAR_FUTURE_EPOCH,
+    FAR_FUTURE_EPOCH, EFFECTIVE_BALANCE_INCREMENT,
```
Comment (F4ever) [inline src/utils/validator_state.py:5]: split in two lines

File: tests/utils/test_validator_state.py [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+def test_is_validator_eligible_to_exit(activation_epoch, exit_epoch, epoch, expected):
+    validator = ValidatorFactory.build()
+    validator.validator.activation_epoch = activation_epoch
+    validator.validator.exit_epoch = exit_epoch
+
+    actual = is_validator_eligible_to_exit(validator, EpochNumber(epoch))
+    assert actual == expected
+
+
+def validators():
```
Comment (F4ever) [inline tests/utils/test_validator_state.py:102]: if it is not fixture, do
```suggestion
def get_validators():
```

File: src/utils/validator_state.py (hunk 5 above)
Comment (sergeyWh1te) [reply src/utils/validator_state.py]: ebad50a2374017e6b76d650376627ec8f902e9cb

File: src/utils/validator_state.py (hunk 4 above)
Comment (sergeyWh1te) [reply src/utils/validator_state.py]: ebad50a2374017e6b76d650376627ec8f902e9cb

File: src/utils/validator_state.py (hunk 3 above)
Comment (sergeyWh1te) [reply src/utils/validator_state.py]: ebad50a2374017e6b76d650376627ec8f902e9cb

File: src/utils/validator_state.py (hunk 2 above)
Comment (sergeyWh1te) [reply src/utils/validator_state.py:84]: Unfortunately, it's possible here.

Stack:
1. MidtermSlashingPenalty.get_validator_midterm_penalty::172
2. MidtermSlashingPenalty.predict_midterm_penalty_in_frame
3. frames_lido_midterm_penalties = MidtermSlashingPenalty.get_future_midterm_penalty_sum_in_frames(
            blockstamp.ref_epoch, all_slashed_validators,  total_balance, future_frames_lido_validators,
 )
4. total_balance = src/services/bunker_cases/midterm_slashing_penalty.py::58

Screenshot 2023-03-20 at 13 28 22

**Pull request closed without merging**